CMakeLists.txt

```diff
@@ -1095,10 +1073,14 @@ if (ZLIB_ENABLE_TESTS)
         endif()
     endmacro()
 
-    set(OPEN_MODES -f -h -R -F -T)
-    set(COMPRESSION_LEVELS -0 -1 -6 -9)
-    foreach(OPEN_MODE ${OPEN_MODES})
-        foreach(COMPRESSION_LEVEL ${COMPRESSION_LEVELS})
+    set(OPEN_MODES -f -h -R -F -T -A)
+    set(COMPRESSION_LEVELS -0 -1 -2 -3 -4 -6 -7 -9)
+
+    foreach(COMPRESSION_LEVEL ${COMPRESSION_LEVELS})
+        minigzip_stdio_cmp(minigzip "lcet10" "test/data/lcet10.txt" ${COMPRESSION_LEVEL})
+        minigzip_stdio_cmp(minigzip "fireworks" "test/data/fireworks.jpg" ${COMPRESSION_LEVEL})
+        minigzip_stdio_cmp(minigzip "paper-100k" "test/data/paper-100k.pdf" ${COMPRESSION_LEVEL})
+        foreach(OPEN_MODE ${OPEN_MODES})
```

Comment:
Reading the test logs is kind of a mess now, could that be improved by reordering the levels of loops?

Would this be better?
```
for (file)
    for (mode)
        for (level)
```
Not sure how to do two parameters for the file loop though.

The extra non-mode tests above could possibly also be removed by including an empty ("") mode in the list, at least I assume that would work in cmake lists.

Reply:
I have simplified it with the `for (file)` method. I tried using the empty string or even a string with a single space in it, but the way CMake parses arg lists was having difficulty with it.

The ordering should look something like this now:

![image](https://user-images.githubusercontent.com/1364432/83345759-94661400-a2cb-11ea-819b-d9f7d0a94685.png)